src/Serilog.Sinks.Seq/Sinks/Seq/SeqCompactJsonFormatter.cs

```diff
@@ -0,0 +1,164 @@
+    /// <param name="output">The output.</param>
+    public void Format(LogEvent logEvent, TextWriter output)
+    {
+        FormatEvent(logEvent, output, _valueFormatter);
+        output.WriteLine();
+    }
+
+    /// <summary>
+    /// Format the log event into the output.
+    /// </summary>
+    /// <param name="logEvent">The event to format.</param>
+    /// <param name="output">The output.</param>
+    /// <param name="valueFormatter">A value formatter for <see cref="LogEventPropertyValue"/>s on the event.</param>
+    public static void FormatEvent(LogEvent logEvent, TextWriter output, JsonValueFormatter valueFormatter)
+    {
+        if (logEvent == null) throw new ArgumentNullException(nameof(logEvent));
+        if (output == null) throw new ArgumentNullException(nameof(output));
+        if (valueFormatter == null) throw new ArgumentNullException(nameof(valueFormatter));
+
+        output.Write("{\"@t\":\"");
+        output.Write(logEvent.Timestamp.UtcDateTime.ToString("O"));
+        output.Write("\",\"@mt\":");
+        JsonValueFormatter.WriteQuotedJsonString(logEvent.MessageTemplate.Text, output);
+
+        var tokensWithFormat = logEvent.MessageTemplate.Tokens
+            .OfType<PropertyToken>()
+            .Where(pt => pt.Format != null);
+
+        // Better not to allocate an array in the 99.9% of cases where this is false
+        // ReSharper disable once PossibleMultipleEnumeration
+        if (tokensWithFormat.Any())
+        {
+            output.Write(",\"@r\":[");
+            var delim = "";
+            foreach (var r in tokensWithFormat)
+            {
+                output.Write(delim);
+                delim = ",";
+                var space = new StringWriter();
+                r.Render(logEvent.Properties, space, CultureInfo.InvariantCulture);
+                JsonValueFormatter.WriteQuotedJsonString(space.ToString(), output);
+            }
+            output.Write(']');
+        }
+
+        if (logEvent.Level != LogEventLevel.Information)
+        {
+            output.Write(",\"@l\":\"");
+            output.Write(logEvent.Level);
+            output.Write('\"');
+        }
+
+        if (logEvent.Exception != null)
+        {
+            output.Write(",\"@x\":");
+            JsonValueFormatter.WriteQuotedJsonString(logEvent.Exception.ToString(), output);
+        }
+
+        if (logEvent.TraceId != null)
+        {
+            output.Write(",\"@tr\":\"");
+            output.Write(logEvent.TraceId.Value.ToHexString());
+            output.Write('\"');
+        }
+
+        if (logEvent.SpanId != null)
+        {
+            output.Write(",\"@sp\":\"");
+            output.Write(logEvent.SpanId.Value.ToHexString());
+            output.Write('\"');
+        }
+
+        var skipSpanProperties = false;
+        if (logEvent is {TraceId: not null, SpanId: not null} &&
+            logEvent.Properties.TryGetValue("SpanStartTimestamp", out var st) &&
```

Comment:
I've kept to the convention of using string literals in this file, since it makes the code easier to inspect (juggling hard-coded strings is pretty much all this class does :-) )